### DISCUSSION TITLE
Possibility to import scripts

### DISCUSSION THREAD

**Discussion by EmielRous** [gapitio/gapit-htmlgraphics-panel#142] [Q&A]
I was wondering if it is possible to use external scripts inside this plugin?

I've tried adding a `<script>` tag to the html with a `src` attribute pointing to cdn script, but I can't seem to access it

**Comment by ZuperZee** [comment]
In short: No. `<script>` doesn't work in HTMLGraphics, but can be included with a [bundler](https://gapit-htmlgraphics-panel.gapit.io/docs/projects/#bundlers).

There is an open issue on it https://github.com/gapitio/gapit-htmlgraphics-panel/issues/65.
This issue might also be useful https://github.com/gapitio/gapit-htmlgraphics-panel/issues/41. Instead of adding the script to the HTML, it can be bundled into the javascript part.
This issue https://github.com/gapitio/gapit-htmlgraphics-panel/issues/112 is about how to add video.js to HTMLGraphics. Final repo for the video.js part https://github.com/gapitio/htmlgraphics-video.js-example. There are also some more examples at the bottom of https://gapit-htmlgraphics-panel.gapit.io/docs/projects/#bundlers.
Some packages/external scripts don't work in HTMLGraphics when bundled. Mainly those that use the [`document` interface](https://developer.mozilla.org/en-US/docs/Web/API/Document). Like the [neovis package](https://www.npmjs.com/package/neovis.js?activeTab=explore) explained in this issue https://github.com/gapitio/gapit-htmlgraphics-panel/issues/132.
If you say which package you want to add, I could make an example repo for it.

**Reply by EmielRous** [reply]
I would love to use the jsPDF library. I'm trying to make a download as PDF Button

**Reply by ZuperZee** [reply]
![image](https://user-images.githubusercontent.com/39485579/232150436-3fdea5f0-cdc2-4391-8378-609993f09d66.png)

[Example repo](https://github.com/ZuperZee/htmlgraphics-jspdf-example)
[panel-options.json](https://github.com/ZuperZee/htmlgraphics-jspdf-example/blob/5aa9760d4002f947ebaa04042f1b760244fca59b/dist/panel-options.json)

There are some steps on how to create the example in the [README.md](https://github.com/ZuperZee/htmlgraphics-jspdf-example/blob/main/README.md)
